pkg/provider/provider.go

```diff
@@ -736,9 +736,27 @@ func (k *kubeProvider) Delete(
 		Name:   name,
 	}
 
-	err = await.Deletion(config)
-	if err != nil {
-		return nil, err
+	awaitErr := await.Deletion(config)
+	if awaitErr != nil {
+		initErr, isInitErr := awaitErr.(await.InitializationError)
+		if !isInitErr {
+			// Object deletion failed.
```

Comment:
This comment is a little weird, might be better to point out that object deletion failed, but didn't (as far as we know) orphan any objects.